pkg/steps/pod.go

```diff
@@ -191,6 +192,47 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 	}
 }
 
+func (s *podStep) getEnvMap() map[string]string {
```

Comment:
We do vendor `test-infra` in `ci-tools`, can we import and use the function from upstream?

Reply:
Added a `spec` type conversion function.
As a followup, I think we could `import` the type from upstream and reuse it in ours.
Maybe I did not get the point of redefining the spec in our code.

Reply:
I think originally ci-operator did not vendor `test-infra` b/c it's a hell of a vendor. Not sure if that's the reason, but could be.